# General configure
tf_expiry: 0.5                            # [s]
dock_timeout: 240.0                       # [s]
time_for_waiting_dock_frame: 600.0        # [s]
slider_timeout: 15                        # [s] - Waiting time for slider motor reachs max position
controller_frequency: 20.0                # [hz]
# control_duration: 80.0
# gear_ratio: 10.33
# encoder_resolution: 82640.0
# wheel_radius: 0.0625
back_laser_offset: 0.52322                # Distance from base_link to back_laser_link when slider at original position
steer_distance_threshold: 0.6

# Frames
base_link: base_link
charger_link: charger_frame
first_frame: cart_frame
last_frame: last_frame
tag_frame: tag_frame
parallel_frame: parallel_frame
dropoff_frame: m104_dropoff_frame

# Goal threshold
stop_yaw_diff: 0.02           # [rad]
stop_trans_diff: 0.01         # [m]
y_tolerance_pid: 0.015        # [m]
yaw_predock_tolerance: 0.02   # [rad]
max_parallel_offset: 0.01     # [m] - will move to parallel.c if exceeded
predock_tf_samples: 5         # tf samples to avg, parallel.c validation

# Velocity informations
linear_vel_range: [-0.2, 0.2]
angular_vel_range: [-0.5, 0.5]    
max_angular_vel: 0.5                  # [rad/s]
min_angular_vel: 0.1                 # [rad/s]
max_angular_accel: 3.2
max_angular_deccel: 0.3
angle_threshold: 0.35
# max_linear_vel: 0.2
# min_linear_vel: 0.035                 # [m/s]
max_linear_vel: 0.3
min_linear_vel: 0.035                 # [m/s]
max_linear_vel_predock: 0.15                   # [m/s]
max_x_pid_steer: 0.055                # [m/s] - Max pid's linear velocity when use pid controller in steer state
min_x_pid_steer: 0.045                 # [m/s] - Min pid's linear velocity when use pid controller in steer state
max_x_pid_lastmile: 0.04             # [m/s] - Max pid's linear velocity when use pid controller in lastmile state
min_x_pid_lastmile: 0.035              # [m/s] - Min pid's linear velocity when use pid controller in lastmile state
max_x_lastmile: 0.03                  # [m/s] - Max linear velocity in lastmile state
# max_x_with_retry_high_current: 0.1    # [m/s] - Max linear velocity when move with encoder
# max_x_out_dock: 0.2                   # [m/s] - Max linear velocity when go out dock
# max_x_correction: 0.05                # [m/s] - Max linear velocity when use correct_robot function
# max_x_go_out_dock: 0.2                # [m/s] - Max linear velocity when go out dock
# max_x_custom_rotation: 0.1            # [m/s] - Max linear velocity when do custom rotation line 56
# max_angular_accel: 3.0
# angular_vel_lim: 0.3

k_p: 0.8
k_i: 0.0
k_d: 0.01
k_p_steer: 2.5
k_i_steer: 0.0
k_d_steer: 0.01

retry_count: 5                        # How many times to retry
debug_mode: True                      # Print messages for debugging.